### PR TITLE
🌱 Audit patch withOwnedConditions

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:

/area provider/core

### CODE REVIEW

File: internal/controllers/machinehealthcheck/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -300,7 +307,18 @@ func (r *Reconciler) reconcile(ctx context.Context, logger logr.Logger, cluster
 		}
 		errList := []error{}
 		for _, t := range append(healthy, unhealthy...) {
```
Comment (sbueringer) [inline internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:309]: Not new with this PR, but I'm really confused why we are patching all Machines twice (if I'm reading this correctly)

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold
Up to you. This PR is only a step in the right direction. I'm just confused about the current state of the MHC controller

File: internal/controllers/machinehealthcheck/machinehealthcheck_controller.go (hunk 1 above)
Comment (fabriziopandini) [reply internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:309]: My current understanding is that we are patching here only if remediation is not allowed (and we patch both unhealthy and healthy machines).

The other two patch machines are for when remediation is allowed, one for healthy machines, the other for unhealthy.

But if you want we can take a deeper look together

**Comment by fabriziopandini** [comment]
/hold cancel
(we can look into your question async, it seems unrelated from this change)

File: internal/controllers/machinehealthcheck/machinehealthcheck_controller.go (hunk 1 above)
Comment (sbueringer) [reply internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:309]: Let's take a look together later. To me it looks like some of these patches are always no-ops

**Pull request merged**